crates/astria-sequencer-relayer/src/relayer/mod.rs

```diff
@@ -288,6 +303,63 @@ impl Relayer {
     }
 }
 
+#[instrument(skip_all)]
+async fn confirm_sequencer_chain_id(
+    configured_sequencer_chain_id: String,
+    sequencer_cometbft_client: SequencerClient,
+) -> eyre::Result<()> {
+    let span = Span::current();
+
+    let retry_config = tryhard::RetryFutureConfig::new(u32::MAX)
+        .max_delay(Duration::from_secs(30))
+        .exponential_backoff(Duration::from_secs(1))
+        .on_retry(
+            |attempt: u32, next_delay: Option<Duration>, error: &tendermint_rpc::Error| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    parent: &span,
+                    attempt,
+                    wait_duration,
+                    error = %eyre::Report::new(error.clone()),
+                    "failed to fetch sequencer chain id; retrying after backoff",
+                );
+                futures::future::ready(())
+            },
+        );
+
+    let actual_sequencer_chain_id =
+        tryhard::retry_fn(move || fetch_sequencer_chain_id(sequencer_cometbft_client.clone()))
+            .with_config(retry_config)
+            .in_current_span()
+            .await
+            .wrap_err("retry attempts exhausted; bailing")?;
+
+    if actual_sequencer_chain_id == configured_sequencer_chain_id {
+        info!(sequencer_chain_id = %configured_sequencer_chain_id, "confirmed sequencer chain id");
+        return Ok(());
+    }
+    bail!(
+        "mismatch in sequencer chain id, configured id: `{configured_sequencer_chain_id}`, actual \
+         id: `{actual_sequencer_chain_id}`"
+    )
```

Comment:
With `eyre::ensure!` this could become (changed the wording of the error message a bit to make clear where it came from):
```suggestion
    ensure!(
        actual_sequencer_chain_id == configured_sequencer_chain_id,
        "configured sequencer chain ID does not match remote; configured: `{configured_sequencer_chain_id}`, \
         remote: `{actual_sequencer_chain_id}`",
    );
    info!(sequencer_chain_id = %configured_sequencer_chain_id, "confirmed sequencer chain id");
    Ok(())
```

Reply:
Done in [458f2a2](https://github.com/astriaorg/astria/pull/1063/commits/458f2a27ff3a3fe97880a3b049c4359c1078058f), although I used the term "received" rather than "remote" to be consistent with the similar change above.